From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: CoolLoong <1542536763@qq.com>
Date: Thu, 30 May 2024 21:59:46 +0800
Subject: [PATCH] clean protocol use


diff --git a/build.gradle.kts b/build.gradle.kts
index 3b784bb3e9618b44e453271a6b879894df0a95db..a42072b098cbdc9b32003fbe64e8afff7550fa9c 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -14,11 +14,14 @@ val alsoShade: Configuration by configurations.creating
 
 dependencies {
     implementation(project(":jegenerator-api"))
-    implementation("org.cloudburstmc:block-state-updater:1.20.80-SNAPSHOT")
+    implementation("org.cloudburstmc:block-state-updater:1.20.80-SNAPSHOT") {
+        exclude(group = "org.cloudburstmc", module = "nbt") // use allay nbt
+    }
     compileOnly(files("C:\\Users\\15425\\IdeaProjects\\Allay\\Allay-API\\build\\libs\\Allay-API-1.0.0-all.jar"))
     compileOnly("com.github.powernukkitx:powernukkitx:nightly-build")
     compileOnly("org.projectlombok:lombok:1.18.30")
     annotationProcessor("org.projectlombok:lombok:1.18.30")// AllayMC
+    implementation("org.allaymc:NBT:3.0.9")
     implementation("com.fasterxml.jackson.core:jackson-core:2.15.2")
     implementation("com.fasterxml.jackson.core:jackson-databind:2.15.2")
 
diff --git a/src/main/java/org/allaymc/jegenerator/mappings/type/BedrockBlockDefinition.java b/src/main/java/org/allaymc/jegenerator/mappings/type/BedrockBlockDefinition.java
index 741a6846bbe09a9ff934c6198eed4e959723aae0..350302b5e3a70f13a616593708d0f6fd456bab8b 100644
--- a/src/main/java/org/allaymc/jegenerator/mappings/type/BedrockBlockDefinition.java
+++ b/src/main/java/org/allaymc/jegenerator/mappings/type/BedrockBlockDefinition.java
@@ -2,11 +2,9 @@ package org.allaymc.jegenerator.mappings.type;
 
 import lombok.Getter;
 import org.cloudburstmc.nbt.NbtMap;
-import org.cloudburstmc.protocol.bedrock.data.definitions.BlockDefinition;
 
-public class BedrockBlockDefinition implements BlockDefinition {
+public class BedrockBlockDefinition{
     private final int blockHash;
-
     @Getter
     private final NbtMap state;
 
@@ -19,7 +17,6 @@ public class BedrockBlockDefinition implements BlockDefinition {
      *
      * @return blockHash It is not RuntimeID just for implements
      */
-    @Override
     public int getRuntimeId() {
         return blockHash;
     }
